### PR TITLE
[Flink-20999][docs] - adds usage examples to the Kafka Avro Confluent connector format

### PR DESCRIPTION
**Pull request by sv3ndk**
## What is the purpose of the change

* Provide additional examples of how to use the avro-confluent format. In particular, no example of how to serialize the kafka key or use the upsert connector are currently mentioned in the [avro confluent connector documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/table/connectors/formats/avro-confluent.html)

## Brief change log

  - Added an example of a table serializing the kafka value in an avro record registered in the schema registry, while keeping the key in raw format
  - Added an example of how to write to that table
  - Added an example of a table using the schema registry for both the key and value
  - Added an example of using the schema registry together with the kafka upsert connector

Only the English version of the documentation has been updated so far. Once the  examples are approved, I'm happy to copy/paste them to the Chinese version of the documentation, at which point I'll need support for the few sentences that were updated/added.

## Verifying this change

Manually executed `./build_docs.sh -p` to check visually the result

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

File: docs/dev/table/connectors/formats/avro-confluent.md [hunk 1]
```diff
@@ -45,29 +45,132 @@ Dependencies
...
+  'key.format' = 'raw',
+  'key.fields' = 'the_kafka_key',
+
+  'value.format' = 'avro-confluent',
+  'value.avro-confluent.schema-registry.url' = 'http://localhost:8082',
+  'value.fields-include' = 'EXCEPT_KEY'
 )
 {% endhighlight %}
+
+To which  we can write as follows):
```
Comment (wuchong) [inline docs/dev/table/connectors/formats/avro-confluent.md:83]: I don't fully understand this sentence. Do you mean "We can write data into the kafka table as follows" ?

**Review by wuchong — COMMENTED** [review]
Great examples! Thanks @sv3ndk , I think this would be very useful for users to get started.

File: docs/dev/table/connectors/formats/avro-confluent.md (hunk 1 above)
Comment (sv3ndk) [reply docs/dev/table/connectors/formats/avro-confluent.md:83]: Thanks for the suggestion. Yes, indeed, that's what I meant. 
I'll add a commit to use your version, it's clearer.

**Comment by sv3ndk** [comment]
Thanks for the feed-back @wuchong . 
I copied the code examples to the Chinese version of the documentation, although they contain some English sentences at the moment, which I'm not capable of translating. Please let me know if I should open a ticket for that part or if we can handle it as part of this PR.

File: docs/dev/table/connectors/formats/avro-confluent.zh.md [hunk 2]
```diff
@@ -52,20 +52,126 @@ Avro Schema Registry 格式只能与[Apache Kafka SQL连接器]({% link dev/tabl
 
 <div class="codetabs" markdown="1">
 <div data-lang="SQL" markdown="1">
+
+||TODO: TRANSLATION:||
```
Comment (wuchong) [inline docs/dev/table/connectors/formats/avro-confluent.zh.md:56]: We dont' need the tag. It's fine just place English content in Chinese docs.

File: docs/dev/table/connectors/formats/avro-confluent.zh.md [hunk 3]
```diff
@@ -35,7 +35,7 @@ Avro Schema Registry (``avro-confluent``) 格式能让你读取被 ``io.confluen
 
 当以这种格式写入（序列化）记录时，Avro schema 是从 table schema 中推断出来的，并会用来检索要与数据一起编码的 schema id。我们会在配置的 Confluent Schema Registry 中配置的 [subject](https://docs.confluent.io/current/schema-registry/index.html#schemas-subjects-and-topics) 下，检索 schema id。subject 通过 `avro-confluent.schema-registry.subject` 参数来制定。
 
-Avro Schema Registry 格式只能与[Apache Kafka SQL连接器]({% link dev/table/connectors/kafka.zh.md %})结合使用。
+Avro Schema Registry 格式只能与[Apache Kafka SQL连接器]({% link dev/table/connectors/kafka.zh.md %})结合使用。||TODO: TRANSLATION:|| or the [Upsert Kafka SQL Connector]({% link dev/table/connectors/upsert-kafka.zh.md %}).
```
Comment (wuchong) [inline docs/dev/table/connectors/formats/avro-confluent.zh.md:38]: We can just repalce the whole sentence use the new English sentence. 

```
The Avro Schema Registry format can only be used in conjunction with the [Apache Kafka SQL connector]({% link dev/table/connectors/kafka.zh.md %}) or the [Upsert Kafka SQL Connector]({% link dev/table/connectors/upsert-kafka.zh.md %}).
```

**Review by wuchong — COMMENTED** [review]
Thanks for updating Chinese part.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**